bool execute
---
float64 z
float64 d
